index.js

```diff
@@ -98,17 +98,28 @@ function register (server, options, next) {
     var level
     var found = false
 
+    var logObject
+    if (options.mergeHapiLogData) {
```

Comment:
can you move this to its own variable declared outside of this function? Let's avoid the data lookup.